### DISCUSSION TITLE
ASCII Particle Engine

### DISCUSSION THREAD

**Discussion by kiedtl** [janet-lang/janet#1048] [Show and tell]
Since the beginning of November, I've been working on an ASCII particle engine for my 2D roguelike game. The engine is embedded into Zig and is completely decoupled from the game engine, providing only the animations.

You can see the source [here](https://github.com/kiedtl/roguelike/blob/master/data/scripts/particles.janet), and I wrote a little about my implementation in [this writeup](https://tilde.team/~kiedtl/blog/particles/).

Cheers!

**Comment by sogaiu** [comment]
Thanks for sharing!

The animations are nice -- hope you don't mind putting one of them here:

![condense](https://user-images.githubusercontent.com/33044872/205469118-45a48e6a-6fdf-44cd-a981-4f3f39ad10b9.gif)
